### PR TITLE
Unity Support

### PR DESCRIPTION
**Pull request by ndesai-newrelic**
1) Unity Custom Exception
2) Unity Support

### CODE REVIEW

**Review by ywang-nr — APPROVED** [review]

File: agent-core/src/main/java/com/newrelic/agent/android/Agent.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 public class Agent {
     public static final String VERSION = "#VERSION#";
     public static final String MONO_INSTRUMENTATION_FLAG = "#MONO_INSTRUMENTATION_FLAG#";
+    public static final String UNITY_INSTRUMENTATION_FLAG = "#UNITY_INSTRUMENTATION_FLAG#";
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/Agent.java:20]: How is "_Unity Instrumented_" different than Mono? Is Mono still used? If not remove, or reprovision that? What other agent would use either flag?

File: agent-core/src/main/java/com/newrelic/agent/android/Agent.java [hunk 2]
```diff
@@ -211,4 +212,8 @@ public static boolean isInstantApp() {
         return getImpl().isInstantApp();
     }
 
+    public static String getUnityInstrumentationFlag() {
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/Agent.java:215]: Who/what calls this?

File: agent-core/src/main/java/com/newrelic/agent/android/ApplicationFramework.java [hunk 3]
```diff
@@ -12,5 +12,7 @@ public enum ApplicationFramework {
     Flutter,
     Capacitor,
     Xamarin,
-    MAUI
+    MAUI,
+
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/ApplicationFramework.java:16]: Remove blank line, and make it caps-case.

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]
I feel like since Unity was the only hybrid platform we supported in the past, this seems redundant. Or was it all removed during OSS migration?

Please move to agent-core/src/main/java/com/newrelic/agent/android/hybrid/unity

File: agent-core/src/main/java/com/newrelic/agent/android/Agent.java (hunk 2 above)
Comment (ndesai-newrelic) [reply agent-core/src/main/java/com/newrelic/agent/android/Agent.java:215]: removed this

File: agent-core/src/main/java/com/newrelic/agent/android/ApplicationFramework.java (hunk 3 above)
Comment (cthomas-newrelic) [reply agent-core/src/main/java/com/newrelic/agent/android/ApplicationFramework.java:16]: Should `Maui`. It's not an acronym is it?

File: agent-core/src/main/java/com/newrelic/agent/android/Agent.java (hunk 1 above)
Comment (cthomas-newrelic) [reply agent-core/src/main/java/com/newrelic/agent/android/Agent.java:20]: I still don't see an answer for this. The argument for using both is we have another hybrid target that is also Mono-based. MONO_INSTRUMENTATION_FLAG was added explicitly for Unity (not Xamarin), which is why it's redundant.

Also, it doesn't appear the build files were modified for `UNITY_INSTRUMENTATION_FLAG`, so simply adding another flag isn't enough.

File: gradle.properties [hunk 4]
```diff
@@ -9,7 +9,7 @@ android.useAndroidX=true
 android.disableAutomaticComponentCreation=true
 
 # Version of the SDK to be built and deployed
-newrelic.agent.version = 7.0.0
+newrelic.agent.version = 7.0.1
```
Comment (cthomas-newrelic) [inline gradle.properties:12]: Don't bump the version in a branch, unless it's a `release` branch or explicitly targeted for a release. There won't be a 7.0.1 release, btw.

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]
Doesn't seem much different than the previous review.  Any code you see in the agent relating to Mono was for Unity - why its is redundant. With probably a few exceptions, when you restored MONO support to the agent you were in effect restoring Unity support. 

If MONO is too general to be used for hybrid implementations, then remove it rather than Unity references.

File: gradle.properties (hunk 4 above)
Comment (ndesai-newrelic) [reply gradle.properties:12]: i did these changes to test sonatype for Unity changes.

File: agent-core/src/main/java/com/newrelic/agent/android/Agent.java (hunk 1 above)
Comment (ndesai-newrelic) [reply agent-core/src/main/java/com/newrelic/agent/android/Agent.java]: @cthomas-newrelic you are right, those changes are not required.

File: gradle.properties (hunk 4 above)
Comment (cthomas-newrelic) [reply gradle.properties:12]: Yea, I figured as much. I also accidentally commit the properties file

**Review by cthomas-newrelic — APPROVED** [review]
So now, is the MONO flag still needed, even for Xamarin/Maui?

**Pull request merged**